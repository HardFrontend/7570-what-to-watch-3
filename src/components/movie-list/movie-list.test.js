import React from 'react';
import renderer from 'react-test-renderer';
import MovieList from './movie-list.jsx';

const filmList = [
  {
    title: `Bohemian rhapsody`,
    imgPoster: `img/bohemian-rhapsody.jpg`,
    genre: `Comedy`,
    releaseDate: 2020,
    imgBg: `img/bg-the-grand-budapest-hotel.jpg`,
    runTime: `1h 50m`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    title: `Bdardjeeling Limited`,
    imgPoster: `img/dardjeeling-limited.jpg`,
    genre: `Horror`,
    releaseDate: 2019,
    imgBg: `img/bg-the-grand-budapest-hotel.jpg`,
    runTime: `1h 50m`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    title: `Fantastic beasts the crimes of grindelwald`,
    genre: `Horror`,
    releaseDate: 2019,
    imgPoster: `img/the-grand-budapest-hotel-poster.jpg`,
    imgBg: `img/bg-the-grand-budapest-hotel.jpg`,
    runTime: `1h 50m`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    title: `Johnny english`,
    imgPoster: `img/johnny-english.jpg`,
    genre: `Horror`,
    releaseDate: 2019,
    imgBg: `img/bg-the-grand-budapest-hotel.jpg`,
    runTime: `1h 50m`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  }
];

it(`Render list of films`, () => {
  const tree = renderer
    .create(<MovieList
      filmList={filmList}
      onMovieCardClick={() => {}}
      onMovieCardHover={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
