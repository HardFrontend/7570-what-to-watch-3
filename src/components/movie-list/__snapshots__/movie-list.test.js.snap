// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render list of films 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Bohemian rhapsody"
        height="175"
        src="img/bohemian-rhapsody.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/dev-component"
      >
        Bohemian rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Bdardjeeling Limited"
        height="175"
        src="img/dardjeeling-limited.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/dev-component"
      >
        Bdardjeeling Limited
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic beasts the crimes of grindelwald"
        height="175"
        src="img/the-grand-budapest-hotel-poster.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/dev-component"
      >
        Fantastic beasts the crimes of grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Johnny english"
        height="175"
        src="img/johnny-english.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/dev-component"
      >
        Johnny english
      </a>
    </h3>
  </article>
</div>
`;
